app/models/workflow.rb

```diff
@@ -29,8 +31,16 @@ def metadata
     super.merge(:object_dependencies => object_dependencies)
   end
 
+  def deletable?
+    requests.any? { |request| !request.finished? } ? false : true
+  end
+
   private
 
+  def validate_deletable
+    throw :abort unless deletable?
```

Comment:
@hsong-rh Why this a throw and not a raise exception. Do callbacks suppress exceptions. If we can have a "WorkflowIsActiveException" we would know why exactly a failure happened.

Reply:
It's recommended by rails doc (https://guides.rubyonrails.org/active_record_callbacks.html#halting-execution) to stop the chain.